form/src/main/java/org/dhis2/form/ui/FormView.kt

```diff
@@ -172,12 +182,128 @@ class FormView : Fragment() {
                 val sections = items?.let {
                     formSectionMapper.mapFromFieldUiModelList(it)
                 } ?: emptyList()
+
+                var resultDialogData: FormViewModel.FormActions.ShowResultDialog? by remember {
+                    mutableStateOf(null)
+                }
+
+                LaunchedEffect(viewModel.actionsChannel) {
+                    viewModel.actionsChannel.collect { action ->
+                        when (action) {
+                            FormViewModel.FormActions.OnFinish ->
+                                onFinishDataEntry?.invoke()
+
+                            is FormViewModel.FormActions.ShowResultDialog ->
+                                resultDialogData = action
+                        }
+                    }
+                }
+
                 Form(
                     sections = sections,
                     intentHandler = ::intentHandler,
                     uiEventHandler = ::uiEventHandler,
                     resources = Injector.provideResourcesManager(context),
                 )
+
+                if (resultDialogData != null) {
+                    val model = resultDialogData!!.model
+                    val allowDiscard = resultDialogData!!.allowDiscard
+                    val fieldsWithIssues = resultDialogData!!.fieldsWithIssues
+
+                    BottomSheetShell(
```

Comment:
What do you think about extracting this to a independent class internal class inside the package? something like: DataEntryBottomSheet ?